### PR TITLE
Reopen same context tab on reload

### PR DESCRIPTION
**Pull request by tikhsuP**
This PR solves issue #512.

### CODE REVIEW

File: scripts/context.js [hunk 1]
```diff
@@ -69,9 +70,15 @@ function singlePageView() {
       } else {
         contexts_dic[feature]()
         $(featureTabId).click(function(event) {
-          openContextFeature(event, featureId)
+          let featureTabId = openContextFeature(event, featureId) + '_tab'
```
[the remark this answers is not in the document]
Comment (tikhsuP) [reply scripts/context.js:73]: I am making it to return the 'feature' parameter, which is further being stored using chrome storage.

[the remark this answers is not in the document]
Comment (anishsarangi) [reply scripts/context.js:73]: Ah... !! I missed the above top line in `changes`.

File: scripts/context.js [hunk 2]
```diff
@@ -42,6 +42,7 @@ function openContextFeature(evt, feature) {
   // Show the current tab, and add an "active" class to the button that opened the tab
   $(feature).show()
   evt.currentTarget.className += ' active'
+  return(feature)
```
Comment (anishsarangi) [inline scripts/context.js:45]: remove brackets

Comment (tikhsuP) [reply scripts/context.js:45]: okay, thanks.

**Comment by anishsarangi** [comment]
In chrome.tabs.onActivated `chrome.tabs.update(id, { url: chrome.runtime.getURL('context.html') + '?url=' + tab.url })` pass the featureID as a URL parameter. AND in the context.js you can get the URL parameter as make that clicked automatically.

**Comment by tikhsuP** [comment]
@anishsarangi I tried a different solution. I was already working on that. Please have a look. 
Thanks :)

File: scripts/context.js [hunk 3]
```diff
@@ -69,16 +69,21 @@ function singlePageView() {
...
         contexts_dic[feature]()
         $(featureTabId).click(function(event) {
-          openContextFeature(event, featureId)
+          let openedFeature = openContextFeature(event, featureId) + '_tab'
+          chrome.storage.sync.set({openedFeature: openedFeature}, function() {
+          })
+        })
+        //Set the default feature to be opened
+        chrome.storage.sync.get(['openedFeature'], function(result) {
+          openedFeature = result.openedFeature
```
Comment (anishsarangi) [inline scripts/context.js:78]: Can we rename this variable ? And please initialise the variable properly. Thank you

Comment (anishsarangi) [inline scripts/context.js:78]: Can we rename this variable ? Then same named variable is used in line no-72 And please initialise the variable properly. Thank you

**Comment by anishsarangi** [comment]
@tikhsuP Add 'moz-extension://' in the list of excluded URLs in `utils.js` to make it work properly in firefox.

File: scripts/context.js (hunk 3 above)
Comment (tikhsuP) [reply scripts/context.js]: Apologies. Correcting it. Thanks

**Comment by anishsarangi** [comment]
Looks fine :-)

**Comment by cgorringe** [comment]
I did find the rare case where if you, for example, Show All Contexts (in Settings), Show Contexts and select the Annotations tab, then clear Annotations in Settings. Next time you Show Contexts it'll show an empty page with no highlighted tabs. It's not that big of a deal, since the user can just select one of the tabs and it'll work fine.

Looks good, so I'll merge it.

**Pull request merged**